connect/transforms/src/test/java/org/apache/kafka/connect/transforms/ExtractFieldTest.java

```diff
@@ -186,7 +189,98 @@ public void nonExistentNestedFieldWithSchemaShouldFail() {
 
     @Test
     public void testExtractFieldVersionRetrievedFromAppInfoParser() {
-        assertEquals(AppInfoParser.getVersion(), xform.version());
+        assertEquals(AppInfoParser.getVersion(), xformKey.version());
+    }
+
+    @Test
+    @DisplayName("Null key value instead of default must be used")
+    public void whenKeyConverterReplaceNullWithDefaultIsFalseExtractFieldWithNullValueMustReturnNull() {
+
+        Map<String, Object> config = new HashMap<>();
+        config.put("field", "optional_with_default");
+        config.put("replace.null.with.default", false);
+
+        xformKey.configure(config);
+
+        final Schema keySchema = SchemaBuilder.struct()
+                .field("optional_with_default", SchemaBuilder.int32().optional().defaultValue(42).build())
+                .build();
+        final Struct key = new Struct(keySchema).put("optional_with_default", null);
+
+        final SinkRecord record = new SinkRecord("test", 0, keySchema, key, null, null, 0);
+
+        final SinkRecord transformedRecord = xformKey.apply(record);
+        Integer extractedValue = (Integer) transformedRecord.key();
+
+        assertNull(extractedValue);
+    }
+
+    @Test
+    @DisplayName("Default key value must be used")
+    public void whenKeyConverterReplaceNullWithDefaultIsTrueExtractFieldWithNullValueMustReturnDefaultValueWhenPresent() {
```

Comment:
Can we simplify the test name? What about `testUnsetOptionalFieldReplacedWithDefault()`?
Same in the other file below.